Python/Product/TestAdapter/PythonRunSettings.cs

```diff
@@ -0,0 +1,227 @@
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Linq;
+using System.Windows.Threading;
+using System.Xml.XPath;
+using Microsoft.PythonTools.Interpreter;
+using Microsoft.VisualStudio.ComponentModelHost;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.TestWindow.CodeCoverage;
+using Microsoft.VisualStudio.TestWindow.Extensibility;
+using Microsoft.VisualStudioTools;
+using Microsoft.VisualStudioTools.TestAdapter;
+
+namespace Microsoft.PythonTools.TestAdapter {
+
+    /// <summary>
+    /// Configure run settings to disable the normal code coverage plugin.
+    /// 
+    /// This plugin will write out an empty file, and the user will end up seeing that 
+    /// file (sometimes selected by default)  in the code coverage results.  We want to 
+    /// keep the file that we're going to generate for code coverage information instead.
+    /// </summary>
+    [Export(typeof(IRunSettingsService))]
+    class PythonRunSettings : IRunSettingsService {
+        private readonly IComponentModel _compModel;
+        private readonly IServiceProvider _serviceProvider;
+        private readonly Dispatcher _dispatcher;
+        internal static Uri PythonCodeCoverageUri = new Uri("datacollector://Microsoft/PythonCodeCoverage/1.0");
+        internal static string CodeCoverageUriString = @"datacollector://Microsoft/CodeCoverage/2.0";
+        internal const int cmdidImportCoverage = 0x10f;
+        [ImportingConstructor]
+        public PythonRunSettings([Import(typeof(SVsServiceProvider))]IServiceProvider serviceProvider) {
+            _compModel = (IComponentModel)serviceProvider.GetService(typeof(SComponentModel));
+            var opState = _compModel.GetService<IOperationState>();
+            opState.StateChanged += StateChange;
+            _dispatcher = Dispatcher.CurrentDispatcher;
+            _serviceProvider = serviceProvider;
+        }
+
+        private void StateChange(object sender, OperationStateChangedEventArgs e) {
+            if (e.State == TestOperationStates.TestExecutionFinished) {
+                var resultUris = e.Operation.GetRunSettingsDataCollectorResultUri(PythonCodeCoverageUri);
+                if (resultUris != null) {
+                    foreach (var eachAttachment in resultUris) {
+                        string filePath = eachAttachment.LocalPath;
+
+                        if (!string.IsNullOrEmpty(filePath) && File.Exists(filePath)) {
```

Comment:
File.Exists handles null and empty strings just fine - no need for an extra test